### ISSUE TITLE
The command znap restart doesn't respect `$ZDOTDIR` set by `pam`/`systemd`

### ISSUE THREAD

**Issue by AliTajelsir** [marlonrichert/zsh-snap#90]
- Doing `znap restart` reload `zsh` with `$ZDOTDIR` being in `$HOME` when the environment variable is set by `pam` in `~/.pam_environment`:
  ```
  ZDOTDIR DEFAULT=@{HOME}/.config/zsh
  ```
- The command works properly if `$ZDOTDIR` is set in `~/.zshenv`
  ```
  ZDOTDIR=$HOME/.config/zsh
  ```
- Running `zsh 5.8` in `Arch Linux`

**Comment by marlonrichert** [comment]
I have no idea what `pam` is. Can you give me some background info?

**Comment by AliTajelsir** [comment]
Sorry, I have been away from my laptop since yesterday. [Pam](https://wiki.archlinux.org/title/PAM) is Linux Pluggable Authentication Module which provide a framework for system-wide user authentication.

I used to set the environment variables per user through `~/.pam_environment` because it is shell-independent. 
After I reported the bug yesterday I discovered that setting environment variables this way going to be [deprecated](https://github.com/linux-pam/linux-pam/commit/ecd526743a27157c5210b0ce9867c43a2fa27784) soon. Now I'm using [systemd](https://man.archlinux.org/man/environment.d.5) to set the variables (also shell-independent), but the same thing happened.
